lib/core/userinfo/services/identity-group.service.ts

```diff
@@ -0,0 +1,347 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Injectable } from '@angular/core';
+import { Observable, of, from, throwError } from 'rxjs';
+import { catchError, map, switchMap } from 'rxjs/operators';
+import { AppConfigService } from '../../app-config/app-config.service';
+import { AlfrescoApiService } from '../../services/alfresco-api.service';
+import { LogService } from '../../services/log.service';
+import {
+    IdentityGroupSearchParam,
+    IdentityGroupQueryCloudRequestModel,
+    IdentityGroupModel,
+    IdentityGroupQueryResponse,
+    IdentityGroupCountModel
+} from '../models/identity-group.model';
+import { IdentityRoleModel } from '../models/identity-role.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class IdentityGroupService {
+
+    constructor(
+        private alfrescoApiService: AlfrescoApiService,
+        private appConfigService: AppConfigService,
+        private logService: LogService
+    ) {}
+
+    /**
+     * Gets all groups.
+     * @returns Array of group information objects
+     */
+    getGroups(): Observable<IdentityGroupModel[]> {
+        const url = this.getGroupsApi();
+        const httpMethod = 'GET', pathParams = {},
+        queryParams = {}, bodyParam = {}, headerParams = {},
+        formParams = {}, authNames = [], contentTypes = ['application/json'];
+
+        return from(this.alfrescoApiService.getInstance().oauth2Auth.callCustomApi(
```

Comment:
@mauriziovitale is that ok that the API is calling the SSO directly?

Reply:
We don't have any other choice. We need to call an API with the SSO token. In the future, the JS API will expose specific API, I guess.

Reply:
Ok, was just confirming it's expected behavior.